internal/internal_test.go

```diff
@@ -251,73 +252,303 @@ func TestCheckWithLoggerError(t *testing.T) {
 	}
 }
 
-func testAuthzServer(customLogger plugins.Plugin) *envoyExtAuthzGrpcServer {
+func TestConfigValid(t *testing.T) {
+
+	m, err := plugins.New([]byte{}, "test", inmem.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	in := `{"addr": ":9292", "query": "data.test"}`
+	config, err := Validate(m, []byte(in))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if config.Addr != ":9292" {
+		t.Fatalf("Expected address :9292 but got %v", config.Addr)
+	}
+
+	if config.parsedQuery.String() != "data.test" {
+		t.Fatalf("Expected query data.test but got %v", config.Query)
+	}
+}
+
+func TestConfigValidDefault(t *testing.T) {
+
+	m, err := plugins.New([]byte{}, "test", inmem.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	config, err := Validate(m, []byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if config.Addr != defaultAddr {
+		t.Fatalf("Expected address %v but got %v", defaultAddr, config.Addr)
+	}
+
+	if config.parsedQuery.String() != defaultQuery {
+		t.Fatalf("Expected query %v but got %v", defaultQuery, config.parsedQuery.String())
+	}
+}
+
+func TestCheckAllowObjectDecision(t *testing.T) {
+
+	// Example Mixer Check Request for input:
+	// curl --user  bob:password  -o /dev/null -s -w "%{http_code}\n" http://${GATEWAY_URL}/api/v1/products
+
+	var req ext_authz.CheckRequest
+	if err := util.Unmarshal([]byte(exampleAllowedRequest), &req); err != nil {
+		panic(err)
+	}
+
+	server := testAuthzServerwithObjectDecision(&testPlugin{})
 	ctx := context.Background()
+	output, err := server.Check(ctx, &req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if output.Status.Code != int32(google_rpc.OK) {
+		t.Fatalf("Expected request to be allowed but got: %v", output)
+	}
 
-	// Define a RBAC policy to allow or deny requests based on user roles
-	module := `
-	package istio.authz
+	response := output.GetOkResponse()
+	if response == nil {
+		t.Fatal("Expected OkHttpResponse struct but got nil")
+	}
 
-	import input.attributes.request.http as http_request
+	headers := response.GetHeaders()
+	if len(headers) != 2 {
+		t.Fatalf("Expected two headers but got %v", len(headers))
+	}
+
+	expectedHeaders := make(map[string]string)
+	expectedHeaders["x"] = "hello"
+	expectedHeaders["y"] = "world"
 
-	default allow = false
+	assertHeaders(t, headers, expectedHeaders)
+}
+
+func TestCheckDenyObjectDecision(t *testing.T) {
 
-	allow {
-		roles_for_user[r]
-		required_roles[r]
+	exampleRequest := `{
```

Comment:
Why does this test implement it's own input data, when there is the `exampleDeniedRequest` data used in all other `Deny` tests?

Reply:
Good point. Updated the test to use the `exampleDeniedRequest` as input.